### DISCUSSION TITLE
Connecting deck.gl multiple layers with own Map-Server

### DISCUSSION THREAD

**Discussion by xyzamd** [apache/superset#18332] [Q&A / Help]
Hey,

We are trying to visualize maps with multiple layers with deck.gl and have only the following parameter to adjust:

![image](https://user-images.githubusercontent.com/92534590/137713325-04e78bd5-67ec-4891-ac6f-6aaed61a8ecd.png)

Is it possible to change or to configure the parameters of deck.gl multiple layers maps with something like a link of own map-server and show its maps as a chart in superset?

Could we build a frame (custom viz plugin) and visualize our maps from our own map-server on this frame?

Environment
browser type and version: Google Chrome
superset version: 0.0.0dev
Thanks!

**Comment by abdullah-mukadam** [comment]
I was recently looking into building a custom deck-gl chart preset for superset. If you want to add a new deck-gl visualisation on the existing map styles, you could try adding one to the current DeckGL chart preset in superset-ui.

Alternatively, you could try making a DeckGL visualisation from scratch. If you want to add other maps from your own map-server (and not mapbox, which is what superset uses by default), then you will have to make a new viz from from scratch.
You can find steps to make a custom viz on superset here : https://superset.apache.org/docs/installation/building-custom-viz-plugins 

A more robust method of making custom viz charts in superset can be found here: https://www.youtube.com/watch?v=HNkPQtfzXK0 

Link to the current deck-gl visualisations in superset (legacy code) : https://github.com/apache-superset/superset-ui-plugins-deckgl
